sympy/solvers/solveset.py

```diff
@@ -978,6 +985,365 @@ def _solveset(f, symbol, domain, _check=False):
+    to that class. The value returned from this helper should be handled
+    properly in the main (\_transolve) function itself. It could be either
+    the exact solution or a reduced form of the equation which
+    `solveset` can handle.
+
+    Apart from this, few other things needs to be taken care while adding
+    an equation solver:
+
+    - Naming conventions:
+      Name of the identification helper should be like  `\_check\_class`
+      where `class` will be the name or abbreviation of the class of
+      equation. The solving helper will be named as `\_class\_solver`.
+      For example: for exponential equation it becomes `\_check\_expo`
+      and `\_expo\_solver`.
+    - Be sure to consider consider corner cases.
+    - Adding tests for each helper.
+    - Adding a docstring to your helper that describes the method
+      implemented with its proof of correctness by explaining with few
+      examples. Since most of the methods devised to identify and solve
+      particular class of equations are heuristics therefore it
+      becomes necessary to provide proof of correctness of these methods.
+      Following things needs to be included while writing the
+      documentation for the helpers:
+
+      - What is the purpose of the helper.
+      - How it solves the equation and how does it identifies the equation.
+      - Examples should be included to prove its correctness.
+      - What are the input parameters and what does the helper returns.
+    """
+
+    def add_type(eq, symbol, domain):
+        """
+        Helper for `_transolve` to handle equations of
+        `Add` type.
+        """
+        result = ConditionSet(symbol, Eq(eq, 0), domain)
+        # trying to convert to P.O.S form
+        simplified_equation = factor(powdenest(eq))
+
+        if simplified_equation.is_Mul:
+            result = _solveset(
+                simplified_equation, symbol, domain)
+
+        # check if it is exponential type equation
+        elif _check_expo(simplified_equation, symbol):
+            new_eq = _expo_solver(simplified_equation)
+            if new_eq:
+                result = _solveset(new_eq, symbol, domain)
+
+        return result
+
+    def pow_type(eq, symbol, domain):
+        """
+        Helper for `_transolve` to handle equations of
+        power type.
+        """
+        result = ConditionSet(symbol, Eq(eq, 0), domain)
+        new_eq = _expo_solver(eq)
+        if new_eq:
+            result = _solveset(new_eq, symbol, domain)
+
+        return result
+
+    # invert_complex handles the call to the desired inverter based
+    # on the domain specified.
+    lhs, rhs_s = invert_complex(f, 0, symbol, domain)
+
+    unsolved_result = ConditionSet(symbol, Eq(f, 0), domain)
+    result = unsolved_result
+
+    if lhs == symbol or not isinstance(rhs_s, FiniteSet):
+        return rhs_s
+
+    for equation in [lhs - rhs for rhs in rhs_s]:
+        if equation == f:
+            if lhs.is_Add:
```

Comment:
I have created a helper for this case as well as for below case (I don't think there could be more), so that these portion of `transolve` does not get affected when adding new solvers or fixing issues. All the modifications is now restricted to the two internal functions above.